tests/widgets/test_colour_text.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2017  Lucas Hoffmann
+# This file is released under the GNU GPL, version 3 or a later revision.
+# For further details see the COPYING file
+
+import unittest
+from urwid import AttrSpec
+from unittest import mock
+
+from alot.settings.const import settings
+from alot.settings import theme
+from alot.widgets.colour_text import parse_quotes
+
+
+DUMMY_THEME = """\
+[bufferlist]
+    line = '', '', '', '', '', ''
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[envelope]
+    body = '', '', '', '', '', ''
+    header = '', '', '', '', '', ''
+    header_key = '', '', '', '', '', ''
+    header_value = '', '', '', '', '', ''
+[global]
+    body = '', '', '', '', '', ''
+    footer = '', '', '', '', '', ''
+    notify_error = '', '', '', '', '', ''
+    notify_normal = '', '', '', '', '', ''
+    prompt = '', '', '', '', '', ''
+    tag = '', '', '', '', '', ''
+    tag_focus = '', '', '', '', '', ''
+[help]
+    section = '', '', '', '', '', ''
+    text = '', '', '', '', '', ''
+    title = '', '', '', '', '', ''
+[taglist]
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[namedqueries]
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[search]
+    focus = '', '', '', '', '', ''
+    normal = '', '', '', '', '', ''
+    [[threadline]]
+        focus = '', '', '', '', '', ''
+        normal = '', '', '', '', '', ''
+[thread]
+    quote_level_2 = '', '', '', '', '#880', '#886'
```

Comment:
Please use different colors for each value so that no false positive can happen because of color code collision.  I think best would be "recognizable" codes: `#201` and `#202` for level 2, `#301` and `#302` for level 3 ...

Reply:
Fixed in the last push. I also moved the theme variable as global to my script as to avoid reading it for each test case.